src/utils/lbann_library.cpp

```diff
@@ -317,4 +318,149 @@ void print_lbann_configuration(lbann_data::Model *pb_model, lbann_comm *comm, in
   std::cout << std::endl;
 }
 
+void check_mem_capacity(lbann_comm *comm, const std::string sample_list_file, size_t stride, size_t offset) {
```

Comment:
Why is this function defined here? Isn't it specific to JAG?

Reply:
It's specific to conduit, and all data_stores for all readers will be using conduit in the future, so I thought it was general enough to put here. It probably makes more sense to put it in the data_store_jag (which will be renamed data_store_conduit) class.